Library/Homebrew/cask/cmd.rb

```diff
@@ -185,6 +187,14 @@ def self.nice_listing(cask_list)
     def process_options(*args)
       non_options = []
 
+      # remove any non-string arguments that occur when invoking cask cmd's programmatically
+      args.each_with_index.reverse_each do |arg, i|
+        unless arg.is_a? String
```

Comment:
When would this not be a string?

Reply:
When running `brew upgrade <cask>` the cask is loaded by the brew frontend. To avoid reloading the cask, it's passed directly to `Cask::Cmd`

Reply:
> To avoid reloading the cask, it's passed directly to `Cask::Cmd`

I think this is the wrong layer of abstraction then. I think we should call `Cask::Installer` instead of the corresponding `Cask::Cmd`.

Reply:
I'm 100% willing to rewrite this using `Cask::Installer`, but my original impetus for using `Cask::Cmd::Upgrade` was so that it would be the source of truth for cask-related operations, and so that it would match the user's (or at least my) mental model. Not having done too much digging, using `Cask::Installer` without using `Cask::Cmd` would mean the logic related to parsing `HOMEBREW_CASK_OPTS`, and for actually upgrading outdated casks, would have to be duplicated. But feel free to contradict me on this: I see the pain points of using `Cask::Cmd` and would like to avoid using it.

I think I recall Mike mentioning that a long-term goal was to have cask use brew's arg-parser? I suppose a long-term solution then might be to have a class method `def upgrade_casks(casks, args:)`, and have `brew cask upgrade` and `brew upgrade` call that method. Thoughts?

Reply:
@reitermarkus Would you be able to suggest some code changes or, better still if possible, push some changes directly to this PR? Would really like this to land today as it's a pretty bad bug.

Reply:
I think most of the `HOMEBREW_CASK_OPTS` are actually handled in `Cask::Config`, so it should also work with `Cask::Installer`.

I think a shared method like `upgrade_casks` might be the best solution here.

Reply:
Seems like I can't push directly since the PR is from an org repo. I implemented these changes for `upgrade`, should be similar to do for `reinstall` and `uninstall`.

https://github.com/Homebrew/brew/compare/master...reitermarkus:fix-cask-issue-87045

Reply:
@reitermarkus Thanks! Could you cherry-pick this and open (and self-merge) a new PR?

Reply:
(oh you've already cherry-picked! I reckon you can just open a new PR).